src/main/java/com/conveyal/taui/controllers/AggregationAreaController.java

```diff
@@ -99,45 +112,69 @@ public static AggregationArea createAggregationArea (Request req, Response res)
 
         ShapefileReader reader = new ShapefileReader(shpFile);
 
-        List<Geometry> geometries = reader.stream().map(f -> (Geometry) f.getDefaultGeometry()).collect(Collectors.toList());
-        UnaryUnionOp union = new UnaryUnionOp(geometries);
-        Geometry merged = union.union();
+        List<SimpleFeature> features = reader.wgs84Stream().collect(Collectors.toList());
```

Comment:
It seems contradictory to use streams but immediately collect them into a collection. I suppose the original thinking is that a shapefile can be quite large and shouldn't be buffered in memory unless necessary.